### ISSUE TITLE
[Performance]Copying 125000 lines is slow

### ISSUE THREAD

**Issue by 0xABAB** [tmux/tmux#413]
I would expect copying 125000 lines to take less than one second. Instead it takes more than my patience level (and more time than it took to write this message). This is done via the copy-mode of tmux. 

It can arguably be optimized by at least a factor of 60. 

This was tmux version 2.1 on El Capitan.

**Comment by ThomasAdam** [comment]
Any chance you can get some actual numbers.  How long does it actually take, and when you say copy, is this selecting the lines and yanking the output, or is it done via copy-pipe, etc?

Other than moaning, you're not telling us enough information.

**Comment by nicm** [comment]
I expect it could be optimized a lot.

**Comment by nicm** [comment]
An easy one to start would be to do fewer realloc() in window_copy_copy_line.

**Comment by 0xABAB** [comment]
@ThomasAdam I told you it took more time than writing the message (which is certainly more than 30 seconds). If that's not enough information, I don't really see what else there is to discuss. It seems rather silly to ask for clarification, and that's something I would expect only an engineer from India to ask, not you. 

@nicm Why do you close this issue without any real reason? 

The "copying" I was speaking of is the operation which actually puts it in some memory location such that afterwards it can be pasted again within tmux. I don't know how this is called, but I feel like this is very easy to reproduce.

**Comment by krader1961** [comment]
Given your original comment, @0xABAB, it was reasonable for @ThomasAdam to ask for clarification. I would have also asked how you came up with the "optimized by at least a factor of 60." assertion. Although I'm guessing you based that on the fact the copy took approximately 60 seconds and you feel it should only take one second. You might want reread both of your comments. Had you said "I did X and was surprised it took Y seconds. I would have expected it to take closer to one second. Is that reasonable? Can it be improved?" you would likely have gotten a warmer response. Also, when performance is the issue your system configuration is always relevant. If you tell us you're doing this on an [Arduino Mini](https://www.arduino.cc/en/Products/Compare) based system that's a very different situation than if you're doing it on the newest Apple Macbook Pro.

I can't read @nicm's mind but had I made the same reply he did it would be meant as a hint to anyone (e.g., yourself) bothered enough by this about where to start making changes that would improve the situation. Fast is always better than slow but that doesn't mean the volunteers contributing to this project should spend their limited time improving the speed of a really unusual use case.
